source/scale/high-availability-cluster-based-deployment.rst

```diff
@@ -8,462 +8,414 @@ High availability cluster-based deployment
 
-  # 1GB (reduce this to 512MB if your server has less than 32GB of RAM)
-  maintenance_work_mem = 512MB
-  
-  autovacuum_max_workers = 4
-  autovacuum_vacuum_cost_limit = 500
+Proxy server configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+With the cluster set up, configure a load balancer or proxy server (e.g., NGINX) to distribute traffic across the nodes.
 
-  # If you have more than 32 CPUs on your database server, please set the following options to utilize more CPU for your server:
-  max_worker_processes = 12
-  max_parallel_workers_per_gather = 4
-  max_parallel_workers = 12
-  max_parallel_maintenance_workers = 4
+Example configuration for NGINX:
 
-**Config for Postgres Replica node**
+.. code-block:: bash
 
-Copy all the above settings to the read replica, and modify or add only the below. 
+  upstream backend {
+      server 10.10.10.2:8065;
+      server 10.10.10.4:8065;
+      keepalive 256;
+  }
 
-.. code-block:: sh
-  
-  # If the instance is lower capacity than r5.xlarge, then set it to a lower number. 
-  # Also tune the "MaxOpenConns" setting under the "SqlSettings" of the Mattermost app accordingly. 
-  # Note that "MaxOpenConns" on Mattermost is per data source name.
-  max_connections = 1024
+  server {
+      listen 80;
+      server_name mattermost.example.com;
 
-  # This setting should be 16MB on read nodes, and 32MB on writer nodes
-  work_mem = 16MB
+      location / {
+          proxy_pass http://backend;
+          proxy_cache mattermost_cache;
+          client_max_body_size 100M;
+      }
 
-  # The below settings allow the reader to return query results even when the primary has a write process running, a query conflict. 
-  # This is set to on because of the high volume of write traffic that can prevent the reader from returning query results within the timeout. 
-  # https://www.postgresql.org/docs/current/hot-standby.html#HOT-STANDBY-CONFLICT
-  hot_standby = on
-  hot_standby_feedback = on
+      location ~ /api/v[0-9]+/(users/)?websocket$ {
+          proxy_pass http://backend;
+          proxy_set_header Connection "upgrade";
+      }
+  }
 
 Leader election
-^^^^^^^^^^^^^^^^
+~~~~~~~~~~~~~~~~
 
-A cluster leader election process assigns any scheduled task such as LDAP sync to run on a single node in a multi-node cluster environment.
-
-The process is based on a widely used `bully leader election algorithm <https://en.wikipedia.org/wiki/Bully_algorithm>`__ where the process with the lowest node ID number from amongst the non-failed processes is selected as the leader.
+Configure the leader election process to handle tasks like LDAP synchronization, ensuring only one node executes scheduled tasks at a time.
 
+- **Purpose**: Assigns scheduled tasks (e.g., LDAP sync) to a single node in a multi-node cluster.
+- **Mechanism**: Uses the bully algorithm : https://en.wikipedia.org/wiki/Bully_algorithm to elect a leader. The node with the lowest ID among non-failed processes becomes the leader.
+ 
```

Comment:
There is nothing for the admin to configure here. The `Leader election` section in the old document is purely informative.